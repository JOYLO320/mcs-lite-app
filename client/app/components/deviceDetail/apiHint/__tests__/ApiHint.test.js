import React from 'react';
import { shallow } from 'enzyme';
import R from 'ramda';
import { Code, TabItem } from 'mcs-lite-ui';
import ApiHint from '../ApiHint';

jest.mock('../fetch', () => () => new Promise(resolve => resolve('Fake Code')));

it('should redner <ApiHint />', () => {
  const wrapper = shallow(
    <ApiHint
      deviceId="deviceId"
      deviceKey="deviceKey"
      datachannels={[
        { datachannelId: 'datachannelId', datachannelName: 'datachannelName' },
      ]}
      getMessages={R.identity}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('should handle tab click', (done) => {
  const wrapper = shallow(
    <ApiHint
      deviceId="deviceId"
      deviceKey="deviceKey"
      datachannels={[
        { datachannelId: 'datachannelId', datachannelName: 'datachannelName' },
      ]}
      getMessages={R.identity}
    />,
  );

  const arduinoTab = wrapper.find(TabItem).at(1);
  // After clicking
  arduinoTab.simulate('click');
  setTimeout(() => {
    expect(wrapper.find(Code).props()).toMatchSnapshot();
    done();
  }, 100);
});
