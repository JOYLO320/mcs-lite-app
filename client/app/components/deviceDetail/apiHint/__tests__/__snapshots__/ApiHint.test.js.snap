// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle tab click 1`] = `
Object {
  "children": "Fake Code",
  "language": "javascript",
}
`;

exports[`should redner <ApiHint /> 1`] = `
<styled.div>
  <styled.div>
    <div>
      <Heading
        level={4}
      >
        helpful
      </Heading>
      <Styled(Component)>
        example
      </Styled(Component)>
    </div>
    <select
      onChange={[Function]}
      value="datachannelId"
    >
      <option
        value="datachannelId"
      >
        datachannelName
      </option>
    </select>
  </styled.div>
  <Styled(Hr) />
  <styled.div>
    <div>
      datachannelName
      datachannelName
    </div>
    <div>
      datachannelId
      datachannelId
    </div>
    <div>
      apiType
      <styled.span>
        <span>
          <input
            checked={false}
            id="upload"
            name="upload"
            onChange={[Function]}
            type="radio"
          />
          <label
            htmlFor="upload"
          >
            upload
          </label>
        </span>
        <span>
          <input
            checked={true}
            id="retrieve"
            name="retrieve"
            onChange={[Function]}
            type="radio"
          />
          <label
            htmlFor="retrieve"
          >
            retrieve
          </label>
        </span>
      </styled.span>
    </div>
  </styled.div>
  <div>
    <TabItem
      active={true}
      color="primary"
      onClick={[Function]}
      value="javascript"
    >
      Javascript
    </TabItem>
    <TabItem
      active={false}
      color="primary"
      onClick={[Function]}
      value="arduino"
    >
      Arduino
    </TabItem>
    <TabItem
      active={false}
      color="primary"
      onClick={[Function]}
      value="talnet"
    >
      Talnet
    </TabItem>
  </div>
  <styled.div>
    <Styled(CopyButton)
      text=""
    >
      copy
    </Styled(CopyButton)>
    <Code
      language="javascript"
    />
  </styled.div>
</styled.div>
`;
